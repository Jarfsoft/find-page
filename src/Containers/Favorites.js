import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import getDevices from '../deviceAPI';
import actions from '../Actions';
import Device from '../Components/Device';

function Favorites() {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.users);
  const devices = useSelector((state) => state.devices);
  dispatch(actions.page('Favorites'));
  const favorites = useSelector((state) => state.favorites).filter(
    (f) => f.user_ID === user.userId,
  );
  const setDevices = (data) => {
    dispatch(actions.devices(data));
  };
  useEffect(() => {
    getDevices(setDevices);
  }, []);
  return (
    <div className="App">
      {devices.filter(
        ((d) => favorites.find((f) => f.device_ID === d.id) !== undefined),
      ).map((d) => (
        <Device
          key={d.name}
          imageUrl={d.image_url}
          id={d.id}
          name={d.name}
          price={d.price}
        />
      ))}
    </div>
  );
}

export default Favorites;
