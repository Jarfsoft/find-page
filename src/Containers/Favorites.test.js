import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Favorites from './Favorites';

describe('Favorites render', () => {
  const mockStore = configureStore();
  let store;
  const state = { 
    devices: [],
    favorites: [],
  };

  describe('Snapshot check', () => {
    it('should render Favorites section', () => {
      store = mockStore(state);
      const favorites = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Favorites />
            </BrowserRouter>
          </Provider>
        ).toJSON();
      expect(favorites).toMatchSnapshot();
    });
  });
  describe('Favorites contains', () => {
    it('should have div', () => {
      const favorites = render(
        <Provider store={store}>
          <BrowserRouter>
            <Favorites />
          </BrowserRouter>
        </Provider>
      );
      const div = favorites.container.querySelectorAll('div');
      expect(div).toHaveLength(1);
    });
  });
});