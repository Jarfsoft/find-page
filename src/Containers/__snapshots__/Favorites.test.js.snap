// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites render Snapshot check should render Favorites section 1`] = `
<div
  className="App"
/>
`;
